feature/settings/src/main/java/com/google/jetpackcamera/settings/SettingsViewModel.kt

```diff
@@ -40,14 +41,15 @@ import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 
 private const val TAG = "SettingsViewModel"
-val fpsOptions = setOf(FPS_15, FPS_30, FPS_60)
+private val fpsOptions = setOf(FPS_15, FPS_30, FPS_60)
 
 /**
  * [ViewModel] for [SettingsScreen].
  */
 @HiltViewModel
 class SettingsViewModel @Inject constructor(
     private val settingsRepository: SettingsRepository,
+    private val permissionChecker: PermissionChecker,
```

Comment:
So I tried to research this a little, and from what I can tell, the permissions are supposed to be fully owned by the UI layer. We should treat them as if they are user interaction events, like pressing a button. With that in mind, we should check permissions in the UI (Activity or Screen level) and send them down to the ViewModel, where the ViewModel can generate the UiState that depends on them.

So the `SettingsViewModel` should have something like:
```
class SettingsViewModel {
    fun setAudioPermissionGranted(granted: Boolean)
}
```

or if you need multiple permissions to be communicated to the ViewModel:

```
class SettingsViewModel {
    fun setGrantedPermissions(permissions: Set<String>)
}
```

This will align with [the guidance](https://developer.android.com/topic/architecture/recommendations#viewmodel) that:
![image](https://github.com/user-attachments/assets/35130637-0efd-46ea-90cb-087d06713685)

whereas currently, this code is storing a `Context` (albeit an _Application_ context) in the ViewModel, which violates this guidance.